driver-core/src/main/com/mongodb/LoggerSettings.java

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import com.mongodb.annotations.Immutable;
+
+import java.util.Objects;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * An immutable class representing settings for logging.
+ *
+ * <p>
+ * The driver logs using SLF4J 1.0 API with a root logger of {@code org.mongodb.driver}.
+ * </p>
+ *
+ * @since 4.9
+ */
+@Immutable
+public final class LoggerSettings {
+    private final int maxCommandLoggingDocumentLength;
+    /**
+     * Gets a builder for an instance of {@code LoggerSettings}.
+     * @return the builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Creates a builder instance.
+     *
+     * @param loggerSettings existing LoggerSettings to default the builder settings on.
+     * @return a builder
+     */
+    public static Builder builder(final LoggerSettings loggerSettings) {
+        return builder().applySettings(loggerSettings);
+    }
+
+    /**
+     * A builder for an instance of {@code LoggerSettings}.
+     */
+    public static final class Builder {
+        private int maxCommandLoggingDocumentLength = 1000;
+        private Builder() {
+        }
+
+        /**
+         * Applies the loggerSettings to the builder
+         *
+         * <p>Note: Overwrites all existing settings</p>
+         *
+         * @param loggerSettings the loggerSettings
+         * @return this
+         */
+        public Builder applySettings(final LoggerSettings loggerSettings) {
+            notNull("loggerSettings", loggerSettings);
+            maxCommandLoggingDocumentLength = loggerSettings.maxCommandLoggingDocumentLength;
+            return this;
+        }
+
+        /**
+         * Sets the max command logging document length.
+         *
+         * @param maxCommandLoggingDocumentLength the max command logging document length
+         * @return this
+         */
+        public Builder maxCommandLoggingDocumentLength(final int maxCommandLoggingDocumentLength) {
+            this.maxCommandLoggingDocumentLength = maxCommandLoggingDocumentLength;
+            return this;
+        }
+
+        /**
+         * Build an instance of {@code LoggerSettings}.
+         * @return the logger settings for this builder
+         */
+        public LoggerSettings build() {
+            return new LoggerSettings(this);
+        }
+    }
+
+    /**
+     * Gets the max command logging document length.
+     *
+     * <p>
+     * When the driver logs a command and its reply via the {@code org.mongodb.driver.protocol.command} SFL4J logger, it truncates the
+     * them to the maximum length defined by this setting.
+     * </p>
+     *
+     * <p>
+     * Defaults to 1000 characters.
+     * </p>
+     *
+     * @return the max command logging document length
+     */
+    public int getMaxCommandLoggingDocumentLength() {
```

Comment:
This could be named just `maxDocumentLength` to generalize it, though not sure where else we'd ever apply this.

Reply:
I don't have any other cases for logging a document in mind right now that aren't a command or reply. but we did make the language in the spec vague as to uses, and had it apply to all extJSON:

> Drivers MUST support configuring the maximum logged length for extended JSON documents in log messages.

so I guess in theory it could come up in the future. maybe if decided to log something like a transaction recovery token or a change stream resume token? I did make the name more general in Rust just in case: https://github.com/mongodb/mongo-rust-driver/blob/c641a38ff02cb384a6ceb7d1d39a97bc270cb015/src/client/options/mod.rs#L530-L542

Reply:
Duplicating the "logging" part in the method name seems redundant: what else can this setting be for if not for logging given that the method is on the `LoggerSettings` class? Given that these are not internal methods, I think we should name them `getMaxDocumentLength`/`maxDocumentLength`.

Reply:
Done